File: Move all negative elements to end - GFG/move-all-negative-elements-to-end.py
#User function Template for python3

def segregateElements( arr, n):
    
    neg = [i for i in arr if i < 0]
    # print(l1)
    pos = [i for i in arr if i >= 0]
    # print(l2)
    
    if len(pos) == 0 or len(neg) == 0:
        return
    
    for i in range(len(pos)):
        arr[i] = pos[i]
        
    j = 0
    while(j<len(neg)):
        arr[i+j+1] = neg[j]
        j+=1

        

#{ 
#  Driver Code Starts
#Initial Template for Python 3


def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        segregateElements(a, n)
        print(*a)

        T -= 1


if __name__ == "__main__":
    main()





    
# } Driver Code Ends

File: Move all negative elements to end - GFG/move-all-negative-elements-to-end.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
class Solution{
    public:
    void segregateElements(int arr[],int n)
    {
        int pos[n],neg[n];
        int j=0, k=0, t;
        for(int i = 0; i<n;i++){
            if(arr[i] > 0){
                pos[j++] = arr[i];
            } else {
                neg[k++] = arr[i];
            }
        }
        int i = 0;
        t = 0;
        while(i<j){
            arr[i++] = pos[t++];
        }
        t = 0;
        while(t<k){
            arr[i++] = neg[t++];
        }
    }
    void printArr(int arr[], int n){
        for(int i=0;i<n;i++){
            printf("%d ",arr[i]);
        }
        printf("\n");
    }
};

// { Driver Code Starts.
int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;
		int a[n];
		for(int i=0;i<n;i++)
		cin>>a[i];
		Solution ob;
		ob.segregateElements(a,n);
		
        for(int i=0;i<n;i++)
        cout<<a[i]<<" ";
		cout<<endl;
	}
}
  // } Driver Code Ends